app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -116,24 +119,29 @@ class BrowserWebViewClient(
 
     @UiThread
     override fun onPageStarted(webView: WebView, url: String?, favicon: Bitmap?) {
-        try {
-            val navigationList = webView.safeCopyBackForwardList() ?: return
-            webViewClientListener?.navigationStateChanged(WebViewNavigationState(navigationList))
-            if (url != null && url == lastPageStarted) {
-                webViewClientListener?.pageRefreshed(url)
-            }
-            lastPageStarted = url
-        } catch (e: Throwable) {
-            GlobalScope.launch {
-                uncaughtExceptionRepository.recordUncaughtException(e, ON_PAGE_STARTED)
-                throw e
+        return runBlocking {
```

Comment:
_- why the return statement when this method doesn't return anything_
Good catch, there's no need for that return.

_- can we avoid blocking altogether?_
I did some changes and we don't need to use `runBlocking` now.
It was necessary because we were calling a coroutine builder inside `LoginDetector`, basically `withContext` to inject javascript in the main thread.
I removed that code and added `@UiThread` annotations instead. Now the responsibility of calling that logic from the main thread is moved to the caller class.

_- why runBlocking over the entire method rather than the coroutine_
If I'm not wrong, using `runBlocking` or not doesn't change "anything" except that it allows me to run suspendable code inside it.

1. Wrapping the entire method vs only the coroutine with `runBlocking`: the result will be the same. Imagine only blocking the coroutine part, when we get to that point of using `runBlocking`, execution will be blocked till the internal suspendable code finishes, and then it will continue. Having `runBlocking` at the top has the same result: `onPageStarted` will not finish till code inside `runBlocking` finishes.

2. Using `runBlocking` vs not using it: The same result, `onPageStarted` will finish when all the logic inside the method finishes.